core/src/test/java/org/opensearch/sql/analysis/AnalyzerTest.java

```diff
@@ -1007,4 +1024,119 @@ public void show_catalogs() {
 
   }
 
+  @Test
+  public void ml_relation_unsupported_action() {
+    Map<String, Literal> argumentMap = new HashMap<>() {{
+        put(ACTION, new Literal("unsupported", DataType.STRING));
+        put(ALGO, new Literal(KMEANS, DataType.STRING));
+      }};
+
+    IllegalArgumentException exception =
+            assertThrows(
+                    IllegalArgumentException.class,
+                    () -> analyze(AstDSL.project(
+                            new ML(AstDSL.relation("schema"), argumentMap), AstDSL.allFields())));
+    assertEquals(
+            "Action error. Please indicate train, predict or trainandpredict.",
+            exception.getMessage());
+  }
+
+  @Test
+  public void ml_relation_unsupported_algorithm() {
+    Map<String, Literal> argumentMap = new HashMap<>() {{
+        put(ACTION, new Literal(PREDICT, DataType.STRING));
+        put(ALGO, new Literal("unsupported", DataType.STRING));
+      }};
+
+    IllegalArgumentException exception =
+            assertThrows(
+                    IllegalArgumentException.class,
+                    () -> analyze(AstDSL.project(
+                            new ML(AstDSL.relation("schema"), argumentMap), AstDSL.allFields())));
+    assertEquals(
+            "Unsupported algorithm: unsupported",
+            exception.getMessage());
+  }
+
+  @Test
+  public void ml_relation_train_sync() {
+    Map<String, Literal> argumentMap = new HashMap<>() {{
+        put(ACTION, new Literal(TRAIN, DataType.STRING));
+        put(ALGO, new Literal(KMEANS, DataType.STRING));
+      }};
+
+    LogicalPlan actual = analyze(AstDSL.project(
+            new ML(AstDSL.relation("schema"), argumentMap), AstDSL.allFields()));
+    assertTrue(((LogicalProject) actual).getProjectList().size() >= 2);
+    assertTrue(((LogicalProject) actual).getProjectList()
+            .contains(DSL.named(STATUS, DSL.ref(STATUS, STRING))));
+    assertTrue(((LogicalProject) actual).getProjectList()
+            .contains(DSL.named(MODELID, DSL.ref(MODELID, STRING))));
+  }
+
+  @Test
+  public void ml_relation_train_async() {
+    Map<String, Literal> argumentMap = new HashMap<>() {{
+        put(ACTION, new Literal(TRAIN, DataType.STRING));
+        put(ALGO, new Literal(KMEANS, DataType.STRING));
+        put(ASYNC, new Literal(true, DataType.BOOLEAN));
+      }};
+
+    LogicalPlan actual = analyze(AstDSL.project(
+            new ML(AstDSL.relation("schema"), argumentMap), AstDSL.allFields()));
+    assertTrue(((LogicalProject) actual).getProjectList().size() >= 2);
```

Comment:
why >= 2, does the result non deterministic?

Reply:
It is deterministic, just at least 2.